### ISSUE TITLE
Accessibility: buttons use colour alone to convey state change

### ISSUE THREAD

**Issue by chris-gds** [tailwindlabs/tailwindui-issues#1507]
## Reproduction URL

https://tailwindui.com/components/application-ui/elements/buttons

![tailwind-buttons](https://github.com/tailwindlabs/tailwindcss/assets/71266765/f134f314-a7a8-4539-81e3-da8380fba994)

---

## Describe your issue

The hover states for buttons uses colour alone to convey a state change (not a WCAG fail)

### Further reading:
https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html

https://govtnz.github.io/web-a11y-guidance/wct/buttons/make-a-button-accessible/buttons-colour-and-contrast.html#:~:text=A%20button%20must%20not%20rely,Knowledge%20Area%3A%20Colour%20and%20contrast.

- Users with partial sight often experience limited color vision.
- Some older users may not be able to see color well.
- Users who have color-blindness benefit when information conveyed by color is available in other visual ways.
- People using text-only, limited color, or monochrome displays may be unable to access color-dependent information.
- Users who have problems distinguishing between colors can look or listen for text cues.
- People using Braille displays or other tactile interfaces can detect text cues by touch.

## Solution

Add an additional visual change to states, border, underline etc.

**Comment by adamwathan** [comment]
Hey! So I double-checked with all of the accessibility experts in my network and none of them have ever heard of it being an issue to only change color on button hover. Everyone agrees that for semantic states like  "liked" (as is mentioned on that site) that you should be sure to communicate it without just color so that color-blind users can still interpret it, but no one I talked to has ever heard of it being important for a button to change size or shape or something on hover.

Even the WCAG site doesn't do anything but change color on hover:

https://github.com/tailwindlabs/tailwindui-issues/assets/4323180/4cdf8df1-b171-4d48-8774-46eb36be6583

So no plans to change anything here out of the box. If you want to change the hover states for your own projects of course you're welcome to, that's one of the big reasons we offer all of these examples as just code snippets rather than a library you install and can't change or anything 👍
